### PR TITLE
prokka 1.13

### PR DESCRIPTION
**Pull request by tseemann**
- [ ] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

**Comment by tseemann** [comment]
```
tar xzf /home/linuxbrew/.cache/Homebrew/prokka-1.13.tar.gz
Error: no implicit conversion of Symbol into String
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/fileutils.rb:1601:in `delete'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/fileutils.rb:1601:in `block in fu_check_options'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/fileutils.rb:1600:in `each'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/fileutils.rb:1600:in `fu_check_options'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.3.3/lib/ruby/2.3.0/fileutils.rb:626:in `rm_r'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/brewsci/homebrew-bio/Formula/prokka.rb:24:in `install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb:153:in `block (2 levels) in install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1095:in `block in brew'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1895:in `block (2 levels) in stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:554:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1894:in `block in stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:119:in `block in unpack'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/extend/fileutils.rb:14:in `block in mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `block in run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `chdir'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/extend/fileutils.rb:74:in `run'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/extend/fileutils.rb:13:in `mktemp'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:114:in `unpack'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/resource.rb:92:in `stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1872:in `stage'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/formula.rb:1090:in `brew'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb:124:in `block in install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/utils.rb:554:in `with_env'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb:121:in `install'
/home/linuxbrew/.linuxbrew/Homebrew/Library/Homebrew/build.rb:202:in `<main>'
```

**Comment by sjackman** [comment]
```
Building and testing XML-SAX-Expat-0.51 ... ! Installing XML::SAX::Expat failed. See /tmp/prokka-20180325-31211-1h2kcbu/prokka-1.13/.brew_home/.cpanm/work/1521966152.31310/build.log for details. Retry with --force to force install it.
! Installing the dependencies failed: Module 'XML::SAX::Expat' is not installed
! Bailing out the installation for XML-Simple-2.25.
FAIL
```
https://travis-ci.org/brewsci/homebrew-bio/builds/357994055#L3857

Possibly…
```ruby
depends_on "expat"
```

**Comment by sjackman** [comment]
CircleCI on macOS
```
Building and testing Net-HTTP-6.17 ... ! Installing Net::HTTP failed. See /private/tmp/prokka-20180325-7426-14autvp/prokka-1.13/.brew_home/.cpanm/work/1521965892.7436/build.log for details. Retry with --force to force install it.
FAIL
```
https://circleci.com/gh/brewsci/homebrew-bio/1602

**Comment by tseemann** [comment]
it seems XML::Simple depends on the whole Perl LWP universe :-(

I have no idea why XML::Sax::Expat is failing - it is a very simpler pure Perl module. It builds fine standalone.

```
uBilding and testing XML-SAX-Expat-0.51 ... ! Installing XML::SAX::Expat failed. See /tmp/prokka-20180325-16900-1dozaka/prokka-1.13/.brew_home/.cpanm/work/1521965999.16968/build.log for details. Retry with --force to force install it.
! Installing the dependencies failed: Module 'XML::SAX::Expat' is not installed
! Bailing out the installation for XML-Simple-2.25.
```

**Comment by sjackman** [comment]
You could debug it interactively on CircleCI and check the `build.log` of `cpanm` for details.

File: Formula/prokka.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  depends_on "perl" unless OS.mac?
+  depends_on "prodigal"
+  depends_on "tbl2asn"
+
+  def install
+    # remove all bundled binaries and use brew ones
+    rm_r "binaries"
+    # install CPAN modules
+    ENV.prepend_create_path "PERL5LIB", prefix/"perl5/lib/perl5"
+    system "cpanm", "--self-contained", "-l", prefix/"perl5",
```
Comment (sjackman) [inline Formula/prokka.rb]: Try removing `--self-contained`. I don't believe it's necessary, and I had trouble with it. I opened an issue upstream with `cpanminus` https://github.com/miyagawa/cpanminus/issues/563

**Comment by tseemann** [comment]
```
Picked up _JAVA_OPTIONS:  -Duser.home=/home/linuxbrew/.cache/Homebrew/java_cache
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: minced has been compiled by a more recent version of the Java Runtime (class file version 53.0), this version of the Java Runtime only recognizes class file versions up to 52.0
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:495)
[08:41:10] Could not determine version of minced - please install version 2.0 or higher
Error: brewsci/bio/prokka: failed
```

**Comment by sjackman** [comment]
`brew test prokka` failed on TravisCI with this error:
```
[08:41:10] Looking for 'minced' - found /home/linuxbrew/.linuxbrew/bin/minced
Picked up _JAVA_OPTIONS:  -Duser.home=/home/linuxbrew/.cache/Homebrew/java_cache
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: minced has been compiled by a more recent version of the Java Runtime (class file version 53.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```
Looks like this error is caused by compiling the byte code with Java 9 and trying to run that JAR file with Java 8.
https://travis-ci.org/brewsci/homebrew-bio/builds/358777220#L3860

File: Formula/prokka.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  depends_on "bioperl"
+  depends_on "blast"
+  depends_on "hmmer"
+  depends_on "infernal"
+  depends_on "less" unless OS.mac?
+  depends_on "minced"
+  depends_on "parallel"
+  depends_on "perl" unless OS.mac?
+  depends_on "prodigal"
+  depends_on "tbl2asn"
```
Comment (sjackman) [inline Formula/prokka.rb]: I like to group the macOS dependencies at the end.
```ruby
unless OS.mac?
  depends_on "less"
  depends_on "perl"
end
```

Comment (sjackman) [reply Formula/prokka.rb]:
```ruby
depends_on :java
```
?

**Comment by sjackman** [comment]
💚 Successful bottles on CircleCI!

File: Formula/prokka.rb (hunk 2 above)
Comment (tseemann) [reply Formula/prokka.rb]: `minced` already depends on java, so i don't need to do it explicitly. it's the only use for it.

**Comment by tseemann** [comment]
```
Error: A JNI error has occurred, please check your installation and try again
Exception in thread "main" java.lang.UnsupportedClassVersionError: minced has been compiled by a more recent version of the Java Runtime (class file version 53.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

Is Java version consistent on all circle mac, circle linux and travis-ci ?

**Comment by sjackman** [comment]
CirlcleCI on Linux does not have Java installed. `depends_on :java` will install the brewed `jdk`, which is Java version 9. (edited)
It looks as though Java 8 is installed on TravisCI (based on the above error).
I'm not sure what version of Java is installed on CircleCI on macOS.

**Comment by tseemann** [comment]
@sjackman 
1. i only got all this to build by removing all the `cpanm` stuff and hacking prokka to avoid using it
2. TRAVIS must use an older Java version, so I can't resolve this
3. I am still pulling it.

**Comment by sjackman** [comment]
@tseemann What did you use instead of `cpanm`?

**Pull request closed without merging**